core/src/main/java/hudson/PluginManager.java

```diff
@@ -1654,24 +1686,86 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
     @Restricted(NoExternalUse.class)
     @RequirePOST public HttpResponse doCheckUpdatesServer() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
+
+        // We'll check the update servers with a try-retry mechanism. The retrier is built with a builder
+        Retrier<FormValidation> updateServerRetrier = new Retrier.Builder<>(
+                // the action to perform
+                this::checkUpdatesServer,
+
+                // the way we know whether this attempt was right or wrong
+                (currentAttempt, result) -> result.kind == FormValidation.Kind.OK,
+
+                // the action name we are trying to perform
+                "check updates server")
+
+                // the number of attempts to try
+                .withAttempts(checkUpdateAttempts)
+
+                // the delay between attempts
+                .withDelay(checkUpdateSleepTimeMillis)
+
+                // whatever exception raised is considered as a fail attempt (all exceptions), not a failure
+                .withDuringActionExceptions(new Class[] {Exception.class})
+
+                // what we do with a failed attempt due to an allowed exception, return an FormValidation.error with the message
+                .withDuringActionExceptionListener( (attempt, e) -> FormValidation.errorWithMarkup(e.getClass().getSimpleName() + ": " + e.getLocalizedMessage()))
+
+                // lets get our retrier object
+                .build();
+
         try {
-            for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
-                FormValidation v = site.updateDirectlyNow(DownloadService.signatureCheck);
-                if (v.kind != FormValidation.Kind.OK) {
-                    // TODO crude but enough for now
-                    return v;
+            // Begin the process
+            FormValidation result = updateServerRetrier.start();
+
+            // Check how it went
+            if (!FormValidation.Kind.OK.equals(result.kind)) {
+                LOGGER.log(Level.SEVERE, Messages.PluginManager_UpdateSiteError(checkUpdateAttempts, result.getMessage()));
+                if (checkUpdateAttempts > 1 && !Logger.getLogger(Retrier.class.getName()).isLoggable(Level.WARNING)) {
```

Comment:
Due to previous log message + this weird call, I am not sure that if block is necessary.

Reply:
I think this second message can be unclear. @MRamonLeon  could you share an example about how it is shown and clarify the purpose of the message? Thanks!